### DISCUSSION TITLE
Semantic Search / RAG with SOLR and Langchain?

### DISCUSSION THREAD

**Discussion by sweitzel** [ciur/papermerge#615] [Ideas]
[RAG ](https://en.wikipedia.org/wiki/Prompt_engineering#Retrieval-augmented_generation) is currently a hot topic and I have successfully experimented with it (ingesting Markdown documents to Elasticsearch and exposing them to a Chat Bot).

As I am now looking for a private Document management system, I would also like to use natural / semantic search over my whole document collection. By using Apache SOLR, the Papermerge projects seems already well prepared for it. However I found no references in regard to RAG/semantic search etc. 

Are there any plans for supporting RAG / semantic search with Papermerge? 
What do you think, would that be a desirable feature for the solution?

**Comment by ciur** [comment]
Yes, semantic search will be added.
